### PR TITLE
task#34- rrefactor the variables to app instance

### PR DESCRIPTION
**Pull request by Dineshjnld**
## Related Issue(s)

[//]:
  #
  "[Cite any related issue(s) or feature request(s) from the repository's issue tracker.]"

Fixes # For #

## Proposed Changes

[//]: # '[List down the specific changes you have made in this pull request.]'

- [x]
- [x]

## Checklist

- [x] I have followed the repository's contribution guidelines.
- [x] I have added necessary documentation or updated existing documentation.
- [x] My code follows the established code style and formatting of the project.
- [x] I have tested my changes and they function as expected.
- [x] I have added suitable unit tests (if applicable).

## To-Do

- [ ]

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -149,9 +148,9 @@ def setDemoMode(val: bool = False):
     print("streamlit rerun")
     st.experimental_rerun()
     print("rerun complete")  # know this
-starter_script: str  # notebook or python_script
+app.starter_script: str  # notebook or python_script
```
Comment (Nasfame) [inline app.py:151]: remove it

File: app.py [hunk 2]
```diff
@@ -149,9 +148,9 @@ def setDemoMode(val: bool = False):
...
 
-temp_dir: str | tempfile.TemporaryDirectory
+app.temp_dir: str | tempfile.TemporaryDirectory
```
Comment (Nasfame) [inline app.py:153]: remove this

File: app.py [hunk 3]
```diff
@@ -165,19 +164,19 @@ def setDemoMode(val: bool = False):
     st.warning("input archive not found: demo:on")
     model_name = "decenter-model-linear-reg-sample_v3"
     input_archive = "samples/sample_v3"
-    temp_dir = "samples/sample_v3"
-    temp_dir_path = temp_dir
+    app.temp_dir = "samples/sample_v3"
+    temp_dir_path = app.temp_dir
```
Comment (Nasfame) [inline app.py:168]: deprecated it to app.work_dir

File: app.py [hunk 4]
```diff
@@ -216,15 +215,15 @@ def setDemoMode(val: bool = False):
             python_repl = os.path.join(venv_dir, "bin", "python3")
 
 driver_scripts = find_driver_scripts(temp_dir_path)
-starter_script = st.selectbox("Training Script:", driver_scripts)
+app.starter_script = st.selectbox("Training Script:", driver_scripts)
 training_cmd: List[str] = None
```
Comment (Nasfame) [inline app.py:219]: = []

File: app.py (hunk 3 above)
Comment (Nasfame) [reply app.py:168]: app.work_dir = replace (....)

**Review by Nasfame — CHANGES_REQUESTED** [review]
...

File: app.py [hunk 5]
```diff
@@ -148,9 +148,9 @@ def setDemoMode(val: bool = False):
     print("streamlit rerun")
     st.experimental_rerun()
     print("rerun complete")  # know this
-app.starter_script: str  # notebook or python_script
+starter_script: str  # notebook or python_script
```
Comment (Nasfame) [inline app.py:151]: why is this even required? I asked u to remove it

File: app.py [hunk 6]
```diff
@@ -164,8 +164,8 @@ def setDemoMode(val: bool = False):
     st.warning("input archive not found: demo:on")
     model_name = "decenter-model-linear-reg-sample_v3"
     input_archive = "samples/sample_v3"
-    app.temp_dir = "samples/sample_v3"
-    temp_dir_path = app.temp_dir
+    app.work_dir = "samples/sample_v3"
+    temp_dir_path = app.work_dir
```
Comment (Nasfame) [inline app.py:168]: why is temp_dir_path required? use app.work_dir everywhere

File: app.py [hunk 7]
```diff
@@ -214,7 +214,7 @@ def setDemoMode(val: bool = False):
         case _:
             python_repl = os.path.join(venv_dir, "bin", "python3")
 
-driver_scripts = find_driver_scripts(temp_dir_path)
+driver_scripts = []
```
Comment (Nasfame) [inline app.py:217]: what nonesense is this? revert this change!!

File: app.py [hunk 8]
```diff
@@ -214,7 +214,7 @@ def setDemoMode(val: bool = False):
...
 app.starter_script = st.selectbox("Training Script:", driver_scripts)
 training_cmd: List[str] = None
```
Comment (Nasfame) [inline app.py:219]: this line should be = []

**Review by Nasfame — CHANGES_REQUESTED** [review]

File: app.py (hunk 8 above)
Comment (Dineshjnld) [reply app.py:219]: which line sir?

Comment (Nasfame) [reply app.py:219]: line no 217 shouldn't be empty. U can see right there that in 218 its referenced, u ill get empty list for driver scripts hence start_script is going to be None

**Review by Nasfame — APPROVED** [review]
LGTM
